scylla/src/transport/connection.rs

```diff
@@ -1564,24 +1564,34 @@ mod tests {
         let (connection, _) = super::open_connection(addr, None, ConnectionConfig::default())
             .await
             .unwrap();
+
         let ks = unique_keyspace_name();
 
-        connection.query_single_page(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}", ks), &[]).await.unwrap();
+        {
+            // Preparation phase
+            let session = SessionBuilder::new()
+                .known_node_addr(addr)
+                .build()
+                .await
+                .unwrap();
+            session.query(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}", ks), &[]).await.unwrap();
+            session
+                .query("DROP TABLE IF EXISTS connection_query_all_tab", &[])
+                .await
+                .unwrap();
```

Comment:
You should do `use_keyspace` before this query, or provide fully qualified names of the keyspaces in queries.